### ISSUE TITLE
git_ignore_path_is_ignored documentation misleading?

### ISSUE THREAD

**Issue by boogisha** [libgit2/libgit2#4720]
Due to an issue experienced elsewhere (https://github.com/git-tfs/git-tfs/issues/1208), I've stumbled across `git_ignore_path_is_ignored()` function from "libgit2", where it returned TRUE for a path already committed to the repository.

I instantly thought this was a bug, as something already tracked shouldn't be subject to ".gitignore" exclude rules (think [`git check-ignore`][1]), but eventually reading `git_ignore_path_is_ignored()` description, this seems to be by design...?

https://github.com/libgit2/libgit2/blob/89c332e41b12a72d89de40d63bc568c56a2c336a/include/git2/ignore.h#L56-L60

I guess there is a good reason for this (like making ignore rules parsing standalone, where one can check for index existence elsewhere... or something), but additional comment that follows seems misleading:

https://github.com/libgit2/libgit2/blob/89c332e41b12a72d89de40d63bc568c56a2c336a/include/git2/ignore.h#L62-L63

This is not correct in the above mentioned case of an already tracked file, as even if it matches a pattern inside ".gitignore", `git add .` will add modifications of that file to index, where `git_ignore_path_is_ignored()` will still return TRUE as if the file is to be ignored (which is not true).

Of course, in case file is untracked, `git add .` will ignore it as expected (not adding it to index), and `git_ignore_path_is_ignored()` returning TRUE would make sense... just that it doesn't align with `git add .` in general, but in some case(s) only.

Should this `git add .` comparison be dropped (or somehow reworded) to avoid confusion, or am I missing something?

Thanks, Buga

[1]: https://git-scm.com/docs/git-check-ignore

**Comment by ethomson** [comment]
> Should this git add . comparison be dropped (or somehow reworded) to avoid confusion, or am I missing something?

Yes, let's reword it to avoid confusion.  The implication is whether `git add .` would add an untracked file, but I appreciate that this is confusing and we should fix it.

**Comment by boogisha** [comment]
Thanks, and yeah, I thought of maybe specifying "untracked" explicitly in there, like:
> ... on the directory containing the **untracked** file, ...

... but I was afraid that would leave an open question of what happens with a _tracked_ file... which then makes the whole point of mentioning `git add .` as an example a bit moot, as it only works the same as `git_ignore_path_is_ignored()` in one case, but not the other, where the other one is even left unclear - does it always work the opposite in case of a tracked file (which would also match an ignore pattern), or it depends (and on what)...?

And going into further details of explaining both cases (and how they differ) seems to add more confusion than benefit :/

(all this in case you meant "reword" as in "keep `git add .`" example)

Otherwise, I don't know if mentioning `git check-ignore --no-index` instead would make more sense, as it seems to be the same behavior as `git_ignore_path_is_ignored()` - or not, as it might not be that well known nor much used command.

**Comment by pks-t** [comment]
I think mentioning `git check-ignore --no-index` instead would be the right thing to do. Do you want to create a pull request for this? :)

**Comment by boogisha** [comment]
Sure :) Let's see...
